pkg/controllers/cluster/cluster_controller.go

```diff
@@ -194,3 +248,155 @@ func (c *Controller) createExecutionSpace(cluster *v1alpha1.Cluster) error {
 
 	return nil
 }
+
+func (c *Controller) monitorClusterHealth() error {
+	clusterList := &v1alpha1.ClusterList{}
+	if err := c.Client.List(context.TODO(), clusterList); err != nil {
+		return err
+	}
+
+	clusters := clusterList.Items
+	for i := range clusters {
+		cluster := &clusters[i]
+		if err := wait.PollImmediate(MonitorRetrySleepTime, MonitorRetrySleepTime*HealthUpdateRetry, func() (bool, error) {
+			// Cluster object may be changed in this function.
+			err := c.tryUpdateClusterHealth(cluster)
+			if err == nil {
+				return true, nil
+			}
+			clusterName := cluster.Name
+			clusterObj := &v1alpha1.Cluster{}
+			if err := c.Get(context.TODO(), client.ObjectKey{Name: clusterName}, clusterObj); err != nil {
```

Comment:
What you want to do is `refresh` the `cluster` object before the next retry by `c.tryUpdateClusterHealth`, right?

Seems the `clusterObj` should be changed to `cluster`.

Reply:
Good idea.